### ISSUE TITLE
Access serial port on Raspberry Pi OS 11

### ISSUE THREAD

**Issue by thetravellor** [zwave-js/zwave-js-ui#3148]
Hello,

I am having issues running zwave-js-ui on RaspberryPi OS 11. 

I have had zwave-js-ui running on this pi for quite some time, it was working fine. Then, I updated the os, which [unfortunately for me] came with an update to the linux kernel:

Linux zigbee 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

This change the serial path of my zooz stick to:

ls -la /dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0-port0
lrwxrwxrwx 1 root root 13 Jun 22 23:29 /dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0-port0 -> ../../ttyUSB0

I am running zwave-js-ui using docker compose:

```
version: '3.7'
services:
  zwavejs2mqtt:
    container_name: zwavejs2mqtt
    image: zwavejs/zwavejs2mqtt:master
    restart: always
    privileged: true
    tty: true
    stop_signal: SIGINT
    environment:
      - SESSION_SECRET=67B4E74235D8E3CFAF4DB0A2E0AAFBC7
      - ZWAVEJS_EXTERNAL_CONFIG=/usr/src/app/store/.config-db
      # Uncomment if you want logs time and dates to match your timezone instead of UTC
      # Available at https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
      - TZ=Australia/Melbourne
    networks:
      - zwave
    devices:
      # Do not use /dev/ttyUSBX serial devices, as those mappings can change over time.
      # Instead, use the /dev/serial/by-id/X serial device for your Z-Wave stick.
      - '/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0-port0:/dev/zwave'
    volumes:
      - ./config/zwave2mqtt/store:/usr/src/app/store
    ports:
      - '8091:8091' # port for web interface
      - '3000:3000' # port for Z-Wave JS websocket server
networks:
  zwave:
```
docker compose up results in

```
zwavejs2mqtt  | 2023-06-22 23:56:49.572 INFO Z-WAVE: Connecting to /dev/zwave
zwavejs2mqtt  | Logging to file:
zwavejs2mqtt  | 	/usr/src/app/store/logs/zwavejs_2023-06-22.log
zwavejs2mqtt  | 2023-06-22 23:56:58.599 DEBUG MQTT: Publishing to zwave/_EVENTS/ZWAVE_GATEWAY-Zwavejs2Mqtt/driver/driver_error: {
zwavejs2mqtt  |   data: [
zwavejs2mqtt  |     ZWaveError: Failed to open the serial port: Error: No such file or directory, cannot open /dev/zwave (ZW0100)
zwavejs2mqtt  |         at Driver.openSerialport (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1226:9)
zwavejs2mqtt  |         at Immediate.<anonymous> (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1061:5) {
zwavejs2mqtt  |       [stack]: 'ZWaveError: Failed to open the serial port: Error: No such file or directory, cannot open /dev/zwave (ZW0100)\n' +
zwavejs2mqtt  |         '    at Driver.openSerialport (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1226:9)\n' +
zwavejs2mqtt  |         '    at Immediate.<anonymous> (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1061:5)',
zwavejs2mqtt  |       message: 'Failed to open the serial port: Error: No such file or directory, cannot open /dev/zwave (ZW0100)',
zwavejs2mqtt  |       code: 100,
zwavejs2mqtt  |       context: undefined,
zwavejs2mqtt  |       transactionSource: undefined,
zwavejs2mqtt  |       name: 'ZWaveError'
zwavejs2mqtt  |     },
zwavejs2mqtt  |     [length]: 1
zwavejs2mqtt  |   ]
zwavejs2mqtt  | } with options { qos: 1, retain: false }
zwavejs2mqtt  | 2023-06-22 23:56:58.602 WARN Z-WAVE: Retry connection in 3 seconds...


```

**Comment by robertsLando** [comment]
@thetravellor tried to use the `/dev/serial/by-id` path instead?

**Comment by thetravellor** [comment]
Does not exist.

**Comment by thetravellor** [comment]
It no longer exists.

```
ls -la /dev/serial
total 0
drwxr-xr-x  3 root root   60 Jun 22 23:29 .
drwxr-xr-x 17 root root 4080 Jun 22 23:29 ..
drwxr-xr-x  2 root root   80 Jun 22 23:29 by-path
pi@zigbee:~/zwave $
```

**Comment by thetravellor** [comment]
I tried the 32 bit kernel also, same result....

uname -a
Linux zigbee 6.1.21-v7l+ #1642 SMP Mon Apr  3 17:22:30 BST 2023 armv7l GNU/Linux

**Comment by robertsLando** [comment]
I have 2 more suggestions so, first is to try using the normal `/dev/ttyACM0` path or whatelse, this should work and in case you will have to disable soft reset. Second one is to use ser2net to push your serial to tcp and then you can connect to it locally through tcp. There must be a reason why that doesn't work btw

**Comment by thetravellor** [comment]
Yep /dev/ttyUSB0:/dev/zwave works

I guess docker is not parsing the raw deice name correctly.

**Comment by thetravellor** [comment]
solved. i guess

**Comment by robertsLando** [comment]
> I guess docker is not parsing the raw deice name correctly.

Could be! Anyway you could have problem with soft reset now, keep that monitored.

I suggest to give a try to this: https://community.home-assistant.io/t/no-dev-serial-by-id-entries-after-latest-update/562195

TLDR; Manually create a symlink with a friendlier name ifrom /serial/by-path to /serial/by-id or wherever you want and use that
